File: Forth_Interpreter/PrintingLine.cpp
#include "PrintingLine.hpp"

#include <ostream>

bool PrintingLine::Run () {
    GeneralDataForExecutors.CommandText.erase(GeneralDataForExecutors.CommandText.begin(),
                                              GeneralDataForExecutors.CommandText.begin() + 3);
    GeneralDataForExecutors.CommandText.pop_back();

    *GeneralDataForExecutors.Output << GeneralDataForExecutors.CommandText << " ";

    return true;
}

namespace ForthFactoryRegistrations {
    FactoryRegistrationCommands<PrintingLine> Command_printLine (".\"");
}

File: Forth_Interpreter_On_Java/PrintingLine/PrintingLine.java
package PrintingLine;

import ForthCommands.ForthCommands;

public class PrintingLine extends ForthCommands{
    
    @Override
    public boolean Run() {
        String command = GeneralDataForExecutors.CommandText;
        System.out.print(command.substring(3, command.length() - 1));

        return true;
    }
}
